### PR TITLE
Add BSC WOOFi trades to dex.trades

### PR DESCRIPTION
**Pull request by LeviScoffie**
Brief comments on the purpose of your changes:

**For Dune Engine V2**
* Added trades.sql, schema.yml, and sources.yml files for trades on BSC. 
* Added project details to the dbt_project.yml file.
* Manually added trades to seeds.csv.
* Updated dex.trades file with ref from trades.sql file.
I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by Hosuke** [comment]
Hi @LeviScoffie , you may have one record in the seed does not match the query result as [shown here](https://github.com/duneanalytics/spellbook/actions/runs/3572698408/jobs/6005886457#step:12:97), please have a look.

**Comment by LeviScoffie** [comment]
Hi @Hosuke trying to have a look and I found that all records in seed csv are consistent with what is in the seed csv file. What could be problem?

**Comment by Hosuke** [comment]
> Hi @Hosuke trying to have a look and I found that all records in seed csv are consistent with what is in the seed csv file. What could be problem?

You may try to modify a query based on [this one](https://dune.com/queries/1675750) and adding conditional filter to filter out the records that match the seeds (having the same unique key). We may find some clues then.

**Comment by Hosuke** [comment]
Hi @LeviScoffie , please have a look at this [query](https://dune.com/queries/1692549?blockchain_t6c1ea=bnb&model_t6c1ea=test_schema.git_d8bb63fb_woofi_bnb_trades&project_t6c1ea=woofi&version_t6c1ea=1&seed_t6c1ea=test_schema.git_d8bb63fb_dex_trades_seed) 
It seems like [this tx](https://bscscan.com/tx/0x2b1f1f9ee3e43dc1bd72619b5b44cf4d9917028aa92ad06d0485a81b3a8f1044) is not from woofi. Please verify this record in the seed file.

File: seeds/dex/trades/dex_trades_seed.csv [hunk 1]
```diff
@@ -119,6 +119,9 @@ avalanche_c,woofi,1,2022-11-08,0xae581cf151eacf9877f628e722514674f3f13b19abcce4e
 avalanche_c,woofi,1,2022-05-10,0x4ab2a9385bda8de7df444e6f342c12dd70ad6300fdf3a351d80f46e119a7811c,122,0x50b7545627a5162f82a992c33b87adc75187b218,4.00000076,0xb97ef9ef8734c71904d8002f8b6bc66dd9c48a6e,126322.419554 
 avalanche_c,woofi,1,2022-07-28,0xa1bcc8d3ec39c8694c2be30f5489091fe1bfe8ffd570841e55e00bef6ee3359c,6,0xb31f66aa3c1e785363f0875a1b74e27b85fd66c7,0.008699999999999999,0xb97ef9ef8734c71904d8002f8b6bc66dd9c48a6e,0.197657
 avalanche_c,woofi,1,2022-09-07,0xf48e0d36d6a51f82edecb408b20c1aa275fddc71f18d4118e734e3116c19d359,8,0xb97ef9ef8734c71904d8002f8b6bc66dd9c48a6e,49.97,0xb31f66aa3c1e785363f0875a1b74e27b85fd66c7,2.714758315778573433
+bnb,woofi,1,2022-11-09,0x2b1f1f9ee3e43dc1bd72619b5b44cf4d9917028aa92ad06d0485a81b3a8f1044,397,0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c,7,0x55d398326f99059ff775485246999027b3197955,4488.067073819728314976
```
Comment (Hosuke) [inline seeds/dex/trades/dex_trades_seed.csv:122]: This record.

**Comment by LeviScoffie** [comment]
Hi @Hosuke I believe the seed records are now consitent with what is in the models

**Review by Hosuke — APPROVED** [review]
To whom it may concern, I have reviewed this PR and here is the summary:

- Metadata & naming conventions look good ✅
- Config blocks match uniqueness tests in schema files ✅
- Tests and seeds added for the new models ✅
- Incremental filters added when joining large tables ✅
- Contract addresses are lowercased ✅ 
- Inner join to large tables(bnb.transactions) as possible ✅

[Sample query results here](https://dune.com/queries/1713435?d=1)

**Review by jeff-dude — APPROVED** [review]

**Comment by jeff-dude** [comment]
thank you @LeviScoffie and @Hosuke -- since woofi already exists on avalance, i went ahead and added model to combine them, prior to dex trades. that should be followed for all protocols cross-chain.

**Pull request merged**